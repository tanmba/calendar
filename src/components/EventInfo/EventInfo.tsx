import React, { MouseEventHandler } from "react";
import { Box } from "@mui/material";

interface EventInfoProps {
  /**
   * Event title.
   */
  title?: string;
  /**
   * Event vendor info.
   */
  vendor?: string;
  /**
   * Event buyer info.
   */
  buyer?: string;
  /**
   * Delete event function.
   */
  deleteEvent?: MouseEventHandler<HTMLButtonElement>;
}

const EventInfo = ({title, vendor, buyer, deleteEvent}: EventInfoProps) => {
  return (
    <React.Fragment>
      <Box component="h2" sx={{ textAlign: "center" }}>Event</Box>
      <p>Title: {title}</p>
      <p>Buyer: {buyer}</p>
      <p>Vendor: {vendor}</p>
      <Box component="button" sx={{ marginTop: "10px" }} onClick={deleteEvent}>
        Delete Event
      </Box>
    </React.Fragment>
  )
}

export default EventInfo;