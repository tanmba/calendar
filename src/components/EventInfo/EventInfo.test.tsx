import EventInfo from "./EventInfo";
import { render } from "@testing-library/react";

describe("Event info component", () => {
  it("renders Event info component correctly", () => {
    const { asFragment } = render(
      <EventInfo
        title="New meeting" 
        vendor="Loewe" 
        buyer="Alice"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});