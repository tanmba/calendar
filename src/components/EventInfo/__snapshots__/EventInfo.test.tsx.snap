// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event info component renders Event info component correctly 1`] = `
<DocumentFragment>
  <h2
    class="MuiBox-root css-xi606m"
  >
    Event
  </h2>
  <p>
    Title: New meeting
  </p>
  <p>
    Buyer: Alice
  </p>
  <p>
    Vendor: Loewe
  </p>
  <button
    class="MuiBox-root css-1q7njkh"
  >
    Delete Event
  </button>
</DocumentFragment>
`;
